gulp/config.js

```diff
@@ -1,83 +1,70 @@
-    tmp: '.tmp/',
+  // where are the entry files of each type
+  // note that this is different from all files.
+  // e.g. /src/elements/button/style.scss
+  // is not an entry file for styles (meaning it shouldn't get compiled)
+  const entryGlobs = {
+    // this will be webpack entries
+    scripts: ['src/scripts/**/*.js'],
+    // this will be compiled as css files
+    styles: ['src/styles/**/*.scss'],
+    // this files will generate pages
+    pages: ['src/pages/**/*.pug'],
+    // this files will generate content files
+    content: ['content/texts/**/*.yml'],
+    assets: [
+      'src/materials/**/',
+      '!src/materials/**/*.{yml,scss,pug,js}',
+      'content/**/*',
+      '!content/texts/**/*',
+      '!content/texts'
+    ],
+    meta: ['src/meta/**/*'],
+    definitions: ['src/**/definition.yml'],
+    tooling: ['gulp/**/*.js', 'gulpfile.js', 'webpack.config.js']
   };
 
-  const languages = ['en'];
-
-  const paths = {
-    scripts: {
-      src: `${basePaths.src}scripts/`,
-      dest: `${basePaths.dest}js/`,
-    },
-    styles: {
-      src: `${basePaths.src}styles/`,
-      dest: `${basePaths.dest}css/`,
-    },
-    content: {
-      src: `${basePaths.content}texts/`,
-      dest: `${basePaths.dest}content/`,
-    },
-    pages: {
-      src: `${basePaths.src}pages/`,
-      dest: basePaths.dest,
-    },
-    layouts: {
-      src: `${basePaths.src}layouts/`,
-    },
-    assets: {
-      src: [
-        `${basePaths.src}materials/icons/`,
-        `${basePaths.src}materials/fonts/`,
-        `${basePaths.src}materials/images/`,
-        `${basePaths.content}images/`
-      ],
-      dest: `${basePaths.dest}assets`,
-    },
-    meta: {
-      src: `${basePaths.src}meta/`,
-      dest: `${basePaths.dest}`,
-    },
-    revManifest: {
-      dest: `${basePaths.dest}rev-manifest.json`,
-    },
+  // Watch globs, whenever this files change, run the corresponding type task.
+  // Note that this is different from the entryGlobs.
+  // e.g. /src/elements/button/style.scss
+  // needs to be watched, but is not an entry file.
+  //
+  // other example is definition files. even though they are not pages,
+  // when they change, they have influence on the final html.
+  const watchGlobs = {
+    scripts: ['src/**/*.js'],
+    styles: ['src/**/*.scss'],
+    pages: ['src/**/*.pug', 'src/**/*.yml'],
+    content: ['content/texts/**/*.yml'],
+    assets: [
+      'src/materials/**/',
+      'content/**/*',
+    ],
+    meta: ['src/meta/**/*']
   };
 
-  const appFiles = {
-    scripts: `${paths.scripts.src}**/*.js`,
-    styles: `${paths.styles.src}**/*.scss`,
-    content: `${paths.content.src}**/*.yml`,
-    pages: `${paths.pages.src}**/*.pug`,
-    layouts: `${paths.layouts.src}**/*.pug`,
-    assets: paths.assets.src.map(folder => `${folder}**/*`),
-    meta: `${paths.meta.src}**/*`,
+  const entryPaths = {
+    src: 'src/',
+    pages: 'src/pages',
+    content: 'content/texts/',
   };
 
-  const components = [
```

Comment:
We'll need those in the new config too.
You removed because you simplified the watch glocb by globing anything in src files but we'll need those folders definition anyway for the styleguide.